v8env/bridge.js

```diff
@@ -0,0 +1,104 @@
+import { logger } from './logger'
+
+const DEFAULT_BRIDGE_TRANSFER_OPTIONS = {
+  release: true,
+  transfer: false
+}
+
+export default function initBridge(ivm, dispatch) {
+  const bridge = global.bridge = {
+    prepareValue(arg, opts = {}) {
+      if (!arg) // false, undefined, null, 0, "" (all transferable)
+        return arg
+
+      let ctor;
+      // this can bomb with certain values
+      try {
+        ctor = arg.constructor
+      } catch (e) {
+        ctor = undefined
+      }
+
+      switch (ctor) {
+        // transferable ivm
+        case ivm.ExternalCopy:
+        case ivm.Reference:
+
+        // primitives
+        case String:
+        case Number:
+        case Boolean:
+          return arg
+
+        // typed arrays
+        case ArrayBuffer:
+        case Uint8Array:
+        case Int8Array:
+        case Uint8ClampedArray:
+        case Int16Array:
+        case Uint16Array:
+        case Int32Array:
+        case Uint32Array:
+        case Float32Array:
+        case Float64Array:
+          return bridge.wrapValue(arg, opts)
+
+        // transferable if wrapped
+        case DataView:
+        case Map:
+        case Set:
+        case RegExp:
+        case Date:
+        case Object: // plain object
+        case Array:
+          return bridge.wrapValue(arg, { release: !!opts.release, transfer: false })
+
+        case Function:
+          return bridge.wrapFunction(arg)
+
+        // simplified-values
+        case Error:
+          return arg.stack || arg.message || arg.toString()
+
+        default:
+          logger.warn("Unknown constructor, wrapping in ivm.Reference:", ctor.name)
```

Comment:
Does it break stuff if this just errors? I _doubt_ wrapping random shit in ivm.Reference is a problem, but if we can just error on stuff we don't recognize that's probably safer.

Reply:
I had that initially as you can see in the commented line below :)

I think it's safe to do either, might be more obvious something is going wrong if it bombs. Good call.